### PR TITLE
ADSECGH-76: Inputs are getting updated from Dictionary

### PR DESCRIPTION
**Pull request by psarras**

### CODE REVIEW

File: AdSecCore/UnitHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      } else if (goo is double number) {
+        if (Quantity.TryFrom(number, units, out var quantity)) {
+          return (T)quantity;
+        }
+      } else if (goo is string txt) {
+        if (double.TryParse(goo.ToString(), out double result)) {
+          if (Quantity.TryFrom(result, units, out var quantity)) {
+            return (T)quantity;
+          }
+        } else if (Quantity.TryParse(typeof(T), txt, out var length)) {
```
Comment (DominikaLos) [inline AdSecCore/UnitHelpers.cs:22]: "out var lenght" pls change it to " out var result" or "out var quantity"

File: AdSecCore/UnitHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          return (T)length;
+        }
+      }
+
+      throw new NotImplementedException();
```
Comment (DominikaLos) [inline AdSecCore/UnitHelpers.cs:27]: maybe better "ArgumentException" ?

File: AdSecCoreTests/ParseQuantityTests.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      Assert.Equal(new Length(2, units), UnitHelpers.ParseToQuantity<Length>("2m", units));
+    }
+
+    [Fact]
+    public void ShouldWorkWithOtherUnits() {
+      Assert.Equal(new Length(2, LengthUnit.Millimeter),
+        UnitHelpers.ParseToQuantity<Length>("2mm", LengthUnit.Millimeter));
+    }
+
+    [Fact]
```
Comment (DominikaLos) [inline AdSecCoreTests/ParseQuantityTests.cs:36]: also add a test for not suported type

File: AdSecGH/Helpers/BusinessExtensions.cs [hunk 4]
```diff
@@ -109,6 +130,36 @@ private static readonly Dictionary<Type, Func<Attribute, object>> ToGoo
...
+            if (goo is double value) {
+              return value;
+            }
+
+            return null;
+          }
+        },
+      };
+
+    public static void UpdateProperties(this IFunction BusinessComponent, GH_Component component) {
```
Comment (DominikaLos) [inline AdSecGH/Helpers/BusinessExtensions.cs:155]: not sure it should be here

File: AdSecGHTests/Helpers/DataConvertorTest.cs [hunk 5]
```diff
@@ -68,19 +68,25 @@ public void ShouldComputeAndAssignOutputs() {
       Assert.NotNull(output.Value);
     }
 
+    [Fact]
+    public void ShouldComputeTheRightResult() {
```
Comment (DominikaLos) [inline AdSecGHTests/Helpers/DataConvertorTest.cs:72]: also add one for unhappy path

File: AdSecCore/UnitHelpers.cs (hunk 1 above)
Comment (psarras) [reply AdSecCore/UnitHelpers.cs:22]: yeah there is already a result there! I will think of something

File: AdSecCore/UnitHelpers.cs (hunk 2 above)
Comment (psarras) [reply AdSecCore/UnitHelpers.cs:27]: I will InvalidCastException! Thanks

File: AdSecCoreTests/ParseQuantityTests.cs (hunk 3 above)
Comment (psarras) [reply AdSecCoreTests/ParseQuantityTests.cs:36]: you mean to throw the exception? done

File: AdSecGH/Helpers/BusinessExtensions.cs (hunk 4 above)
Comment (psarras) [reply AdSecGH/Helpers/BusinessExtensions.cs:155]: anything that is part of IFunction but needs access to GH I've put there.

File: AdSecGHTests/Helpers/DataConvertorTest.cs (hunk 5 above)
Comment (psarras) [reply AdSecGHTests/Helpers/DataConvertorTest.cs:72]: hmm can we think about having conversion tests instead? This suite of tests is testing the default Parameters on a fake component. I've added a test, but it is a bit contrived.

**Review by DominikaLos — APPROVED** [review]

**Pull request merged**